components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/display/DisplayToolbar.kt

```diff
@@ -317,7 +337,15 @@ internal class DisplayToolbar(
         val urlWidth = (width - iconSize - browserActionsWidth - pageActionsWidth -
             menuWidth - navigationActionsWidth - 2 * urlBoxMargin)
         val urlWidthSpec = MeasureSpec.makeMeasureSpec(urlWidth, MeasureSpec.EXACTLY)
-        urlView.measure(urlWidthSpec, fixedHeightSpec)
+
+        if (titleView.isVisible()) {
+            // With a title view, the url and title split the rest of the space vertically
+            titleView.measure(urlWidthSpec, fixedHeightSpec / MEASURED_HEIGHT_DENOMINATOR)
+            urlView.measure(urlWidthSpec, fixedHeightSpec / MEASURED_HEIGHT_DENOMINATOR)
```

Comment:
I haven't fully debugged it yet. But this is the part that looks wrong: `fixedHeightSpec` is a MeasureSpec (packed in an int) and not a height you can divide:
https://developer.android.com/reference/android/view/View.MeasureSpec

Instead you need to create a new spec (see where `fixedHeightSpec` gets created):
```Kotlin
 MeasureSpec.makeMeasureSpec(height / MEASURED_HEIGHT_DENOMINATOR /** Divide here! **/, MeasureSpec.EXACTLY)
```